tests/python/pants_test/backend/project_info/tasks/test_filedeps.py

```diff
@@ -140,99 +147,115 @@ def create_target(path, definition, sources=None):
+          'project/config/app.yaml',
+          'project/BUILD',
+          'src/java/bin/BUILD',
+          'src/java/core/BUILD',
+          'src/java/bin/main.java',
+          'src/java/core/core*.java',
+          'src/java/lib/BUILD',
+          'src/java/lib/lib1.java',
+          'src/resources/lib/*.json',
+          'src/resources/lib/BUILD',
+          'src/scala/core/BUILD',
+          'src/scala/core/core1.scala',
+          'src/thrift/storage/BUILD',
+          'src/thrift/storage/data_types.thrift',
+          targets=[self.target('project:app')],
+          options=dict(globs=True, absolute=is_absolute),
+        )
+
+      def test_scala_java_cycle_scala_end(self, is_absolute=is_absolute):
+        self.assert_console_output(
+          'src/scala/core/BUILD',
+          'src/scala/core/core1.scala',
+          'src/java/core/BUILD',
+          'src/java/core/core1.java',
+          'src/java/core/core2.java',
+          targets=[self.target('src/scala/core')],
+          options=dict(absolute=is_absolute),
+        )
+
+      def test_scala_java_cycle_java_end(self, is_absolute=is_absolute):
+        self.assert_console_output(
+          'src/scala/core/BUILD',
+          'src/scala/core/core1.scala',
+          'src/java/core/BUILD',
+          'src/java/core/core1.java',
+          'src/java/core/core2.java',
+          targets=[self.target('src/java/core')],
+          options=dict(absolute=is_absolute),
+        )
+
+      def test_concrete_only(self, is_absolute=is_absolute):
+        self.assert_console_output(
+          'src/java/lib/BUILD',
+          'src/java/lib/lib1.java',
+          'src/thrift/storage/BUILD',
+          'src/thrift/storage/data_types.thrift',
+          'src/resources/lib/BUILD',
+          'src/resources/lib/data.json',
+          'src/scala/core/BUILD',
+          'src/scala/core/core1.scala',
+          'src/java/core/BUILD',
+          'src/java/core/core1.java',
+          'src/java/core/core2.java',
+          targets=[self.target('src/java/lib')],
+          options=dict(absolute=is_absolute),
+        )
+
+      def test_jvm_app(self, is_absolute=is_absolute):
+        self.assert_console_output(
+          'project/BUILD',
+          'project/config/app.yaml',
+          'src/java/bin/BUILD',
+          'src/java/bin/main.java',
+          'src/java/lib/BUILD',
+          'src/java/lib/lib1.java',
+          'src/thrift/storage/BUILD',
+          'src/thrift/storage/data_types.thrift',
+          'src/resources/lib/BUILD',
+          'src/resources/lib/data.json',
+          'src/scala/core/BUILD',
+          'src/scala/core/core1.scala',
+          'src/java/core/BUILD',
+          'src/java/core/core1.java',
+          'src/java/core/core2.java',
+          targets=[self.target('project:app')],
+          options=dict(absolute=is_absolute),
+        )
+
+      for test in [test_resources, test_globs, test_globs_app, test_scala_java_cycle_scala_end,\
```

Comment:
Rather than hard-coding the test names here (which can be easy to forget to do if you add a test), we could do the slightly nasty:

`for test in [v for k, v in sorted(locals().items()) if k.startswith('test_')]:`

which will automatically add all functions in the local scope whose name starts with `test_`

(In general, `locals()` is a thing you don't want to touch, but here it seems pretty useful)